pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTAgent.java

```diff
@@ -22,19 +23,49 @@ public ModelASTAgent(Object sourceLocation) {
     @Override
     public JSONObject toJSON() {
         final JSONObject j = new JSONObject();
-        j.accumulate("type", agentType.toJSON());
 
-        if (variables != null) {
-            if (variables instanceof ModelASTClosureMap &&
-                    !((ModelASTClosureMap) variables).getVariables().isEmpty()) {
-                j.accumulate("arguments", variables.toJSON());
-            } else if (variables instanceof ModelASTValue) {
-                j.accumulate("argument", variables.toJSON());
+        // Handle JENKINS-43016 - round-trip empty-string label agent into agent any.
+        if (isEmptyStringLabelAgent()) {
+            j.accumulate("type", "any");
+        } else {
+            j.accumulate("type", agentType.toJSON());
+
+            if (variables != null) {
+                if (variables instanceof ModelASTClosureMap &&
+                        !((ModelASTClosureMap) variables).getVariables().isEmpty()) {
+                    j.accumulate("arguments", variables.toJSON());
+                } else if (variables instanceof ModelASTValue) {
+                    j.accumulate("argument", variables.toJSON());
+                }
             }
         }
         return j;
     }
 
+    private boolean isEmptyStringLabelAgent() {
+        if (agentType.getKey().equals("label") || agentType.getKey().equals("node")) {
+            if (variables instanceof ModelASTValue && ((ModelASTValue) variables).getValue().equals("")) {
+                return true;
+            }
+            if (variables instanceof ModelASTClosureMap) {
+                Map<ModelASTKey, ModelASTMethodArg> vars = ((ModelASTClosureMap) variables).getVariables();
+                // Don't actually switch to "agent any" if there are any additional options besides the label.
+                if (vars.size() == 1) {
+                    for (Map.Entry<ModelASTKey,ModelASTMethodArg> entry : vars.entrySet()) {
+                        if (entry.getKey().getKey().equals("label")) {
+                            ModelASTMethodArg argValue = entry.getValue();
+                            if (argValue instanceof ModelASTValue && ((ModelASTValue) argValue).getValue().equals("")) {
```

Comment:
Is there a risk of `getValue()` returning `null`? If so maybe turn around the equals to `"".equals(((ModelASTValue) argValue).getValue())` or `StringUtils.isEmpty(((ModelASTValue) argValue).getValue())` to safeguard against NPE?